library/psa_crypto.c

```diff
@@ -4069,6 +4069,23 @@ static psa_status_t psa_cipher_setup( psa_cipher_operation_t *operation,
                               PSA_KEY_USAGE_ENCRYPT :
                               PSA_KEY_USAGE_DECRYPT );
 
+    status = psa_get_key_from_slot( handle, &slot, usage, alg );
```

Comment:
Because this code path is supposed to support opaque drivers too (which will return an error for `psa_get_transparent_key` but not for `psa_get_key_from_slot`)

Reply:
Yes make sense and as you test before to go to the software fallback that the key is transparent, we can get rid of the call to `psa_get_transparent_key()`, right?

Reply:
Ah, yes. Will remove :+1:

Reply:
Removed.